### ISSUE TITLE
Package m2e tycho extension to smooth import of pde plugins

### ISSUE THREAD

**Issue by gayanper** [testforstephen/vscode-pde#41]
Today when importing pde plugins with latest JDT.LS the plugin projects fails to load. This is due to lack of configurators that required to populate PDE classpath containers and natures.

https://github.com/tesla/m2eclipse-tycho is not actively maintained. It doesn't work with m2e 2.0 as well.

One suggestion is we move required classes related configurators and classpath resolvers into this plugin.

WDYT ?

**Comment by testforstephen** [comment]
> Today when importing pde plugins with latest JDT.LS the plugin projects fails to load. This is due to lack of configurators that required to populate PDE classpath containers and natures.
> 
> https://github.com/tesla/m2eclipse-tycho is not actively maintained. It doesn't work with m2e 2.0 as well.
> 
> One suggestion is we move required classes related configurators and classpath resolvers into this plugin.
> 
> WDYT ?

Yes, m2eclipse-tycho is not compitable with the latest jdt.ls, which depends on m2e@2.0. That's why we removed it from vscode-pde extension.

vscode-pde is developed by us as an internal extension, and used to help team to dogfooding vscode to develop Java language server itself. If you're using vscode-pde to import eclipse.jdt.ls in VS Code, it should work without m2eclipse-tycho.

 What kind of project are you working on with vscode-pde?

**Comment by gayanper** [comment]
I found this when i tried to import vscode java test plugin. And also some of my own plugins i was trying to use vscode to edit.

**Comment by testforstephen** [comment]
> I found this when i tried to import vscode java test plugin. And also some of my own plugins i was trying to use vscode to edit.

I see. Generating the pde classpath and nature into .project and .classpath is a missing part of this pde extension. Since the .project and .classpath structure are similar for all eclipse plugins, a simple workaround is to manually copy an existing .project and .classpath files into your current workspace and reload VS Code.

**Comment by lppedd** [comment]
M2E Tycho support is now merged into the main M2E stream.
https://github.com/eclipse-m2e/m2e-core/pull/967
